mule-module-apikit/src/test/java/org/mule/module/apikit/transform/ApikitResponseTransformerTestCase.java

```diff
@@ -0,0 +1,252 @@
+package org.mule.module.apikit.transform;
+
+import static org.hamcrest.Matchers.instanceOf;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import org.mule.api.MuleContext;
+import org.mule.api.MuleMessage;
+import org.mule.api.registry.MuleRegistry;
+import org.mule.api.registry.RegistrationException;
+import org.mule.api.transformer.Transformer;
+import org.mule.api.transformer.TransformerException;
+import org.mule.module.apikit.exception.ApikitRuntimeException;
+import org.mule.transformer.types.SimpleDataType;
+import org.mule.transport.NullPayload;
+
+import com.google.common.cache.LoadingCache;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.raml.model.MimeType;
+
+/**
+ * Created by alan on 10/30/15.
+ */
+public class ApikitResponseTransformerTestCase
+{
+
+    private ApikitResponseTransformer responseTransformer = new ApikitResponseTransformer();
+
+    @Mock
+    private MuleMessage message;
+    @Mock
+    private MuleContext muleContext;
+    @Mock
+    private MuleRegistry muleRegistry;
+    @Mock
+    private LoadingCache<DataTypePair, Transformer> transformerCache;
+    @Mock
+    private Transformer transformer;
+
+
+    @Before
+    public void setUp() throws ExecutionException, RegistrationException
+    {
+
+        initMocks(this);
+        responseTransformer.setMuleContext(muleContext);
+        when(muleContext.getRegistry()).thenReturn(muleRegistry);
+        when(muleContext.getRegistry().get("__restRouterTransformerCache")).thenReturn(transformerCache);
+
+    }
+
```

Comment:
add tests with a compound accept header, e.g: `Accept:application/json,application/xml;q=0.9,*/*;q=0.8` against three apis, one that supports only json, another for xml and a last one with some other mime type e.g: `text/plain` to verify that `*/*` works as expected